acceptance/tests/vault/vault_tls_auto_reload_test.go

```diff
@@ -0,0 +1,182 @@
+package vault
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"testing"
+	"time"
+
+	terratestLogger "github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/consul"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/helpers"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/k8s"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/logger"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/vault"
+	"github.com/stretchr/testify/require"
+)
+
+// TestVault_TlsAutoReload installs Vault, bootstraps it with secrets, policies, and Kube Auth Method.
+// It then gets certs for https and rpc on the server. It then waits for the certs to rotate and checks
+// that certs have different expirations.
+func TestVault_TlsAutoReload(t *testing.T) {
```

Comment:
This is essentially the TestVault() test, but we make the certificate expiration very short and then we also get HTTPS and RPC certs before and after the time we expect them to expire  to confirm that they have been rotated, as indicated by a new NotAfter value.